src/EurekaClient.js

```diff
@@ -426,10 +430,22 @@ export default class Eureka extends EventEmitter {
   eurekaRequest(opts, callback, retryAttempt = 0) {
     this.clusterResolver.resolveEurekaUrl((err, eurekaUrl) => {
       if (err) return callback(err);
-      const requestOpts = merge({
+      let requestOpts = merge({
         baseUrl: eurekaUrl,
         gzip: true,
       }, opts);
+
+      if (this.requestMiddleware) {
+        try {
+          requestOpts = this.requestMiddleware(requestOpts);
```

Comment:
Haha, @jquatier and I talked about the possibility of people needing that and we can easily do that. I will do a quick update to this PR to make that change. The one caveat when making it async is that it will be evaluated before EVERY eureka request, so I would suggest caching any long lived tokens so you don't make multiple oauth calls.